### PR TITLE
Added support for the Obi keyboard.

### PR DESCRIPTION
**Pull request by ADPenrose**
## Description

Added support for the Obi keyboard: a 40% inspired by the TMOv2 and the Nightmare boards. 

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* As this is the first PR regarding the kintsugi keyboard, no issue has been fixed or closed yet.

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/adpenrose/obi/config.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+#define RGB_DI_PIN F7
+#ifdef RGB_DI_PIN
+#    define RGBLED_NUM 16
+#    define RGBLIGHT_HUE_STEP 8
+#    define RGBLIGHT_SAT_STEP 8
+#    define RGBLIGHT_VAL_STEP 8
+#    define RGBLIGHT_LIMIT_VAL 255 /* The maximum brightness level */
+#    define RGBLIGHT_SLEEP  /* If defined, the RGB lighting will be switched off when the host goes to sleep */
+/*== all animations enable ==*/
+//#    define RGBLIGHT_ANIMATIONS
```
Comment (drashna) [inline keyboards/adpenrose/obi/config.h:46]:
```suggestion
```

File: keyboards/adpenrose/obi/obi.c [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+#include "obi.h"
+
+/* Encoder */
+#ifdef ENCODER_ENABLE
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+    if (!encoder_update_user(index, clockwise)) { 
+        return false; 
+    }
+/* The switch case allows for different encoder mappings on different layers, "default" map gets applied for all unspecified layers */
```
Comment (drashna) [inline keyboards/adpenrose/obi/obi.c:12]: TBH, you want the encoder map feature then. 

https://docs.qmk.fm/#/feature_encoders?id=encoder-map

And yes this works with VIA.
However, encoder map should only be enabled at the keymap level.

File: keyboards/adpenrose/obi/readme.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Obi
+
+![obi](https://i.imgur.com/btQPol1.png)
```
Comment (drashna) [inline keyboards/adpenrose/obi/readme.md:3]:
```suggestion
![obi](https://i.imgur.com/btQPol1h.png)
```

**Review by drashna — CHANGES_REQUESTED** [review]
It looks like you have changes to the submodules here.  These need to be reverted before the PR can be approved. 

Rebasing sometimes will fix this. Otherwise, you'd want to do something like this to fix them:
```sh
git checkout master
git submodule foreach 'git log -n1 --oneline'
# remember the first abcdef1234 on the submodules that have an issue
git checkout @@your-branch@@
# for each of the problematic repo's:
cd lib/@@name@@
git checkout abcdef1234
```

**Comment by ADPenrose** [comment]
I think it's gonna be easier if I just close this PR, delete the branch that is causing problems and then reopen it with these corrections. Thanks.

**Pull request closed without merging**